5-Heroku/server/middlewares.js

```diff
@@ -0,0 +1,14 @@
+const uuid = require("uuid");
+
+module.exports.userIdMiddleware = (req, res, next) => {
+  let userId = req.cookies ? req.cookies.userId : undefined;
```

Comment:
```  
let userId = req.cookies.userId;
  if (!userId) {
    userId = uuid.v4();
    res.cookie("userId", userId);
  }
  req.userId = userId;
  next();
```

Reply:
let userId = req.cookies.userId; throws an error for the first call (cookies is undefined)

Reply:
weird... so use optional chaining 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining